dev/src/bulk-writer.ts

```diff
@@ -351,22 +356,36 @@ export class BulkWriter {
 
   constructor(
     private readonly firestore: Firestore,
-    enableThrottling: boolean
+    private readonly options?: firestore.BulkWriterOptions
   ) {
     this.firestore._incrementBulkWritersCount();
+    validateBulkWriterOptions('options', options);
 
-    if (enableThrottling) {
-      this.rateLimiter = new RateLimiter(
-        STARTING_MAXIMUM_OPS_PER_SECOND,
-        RATE_LIMITER_MULTIPLIER,
-        RATE_LIMITER_MULTIPLIER_MILLIS
-      );
-    } else {
+    if (options?.throttling === false) {
       this.rateLimiter = new RateLimiter(
+        Number.POSITIVE_INFINITY,
         Number.POSITIVE_INFINITY,
         Number.POSITIVE_INFINITY,
         Number.POSITIVE_INFINITY
       );
+    } else {
+      const startingRate =
+        typeof options?.throttling !== 'boolean' &&
+        options?.throttling?.initialOpsPerSecond !== undefined
+          ? options.throttling.initialOpsPerSecond
+          : STARTING_MAXIMUM_OPS_PER_SECOND;
```

Comment:
I think this is:

```suggestion
      const startingRate = options?.throttling?.initialOpsPerSecond ??
          STARTING_MAXIMUM_OPS_PER_SECOND;
```

I tested it with this code:

```
const STARTING_MAXIMUM_OPS_PER_SECOND = 1337;
const options1 : any = {throttling:false};
const options2 : any = {throttling:true};
const options3 : any = {throttling:{initialOpsPerSecond:10}};

function printThrottler(options?:any) {
    console.log(options?.throttling?.initialOpsPerSecond ??STARTING_MAXIMUM_OPS_PER_SECOND)
}

printThrottler(options1);
printThrottler(options2);
printThrottler(options3);
```

This prints 1337, 1337, 10.

Same below.

Reply:
Since it's a union type, `initialOpsPerSecond` is not a number unless I explicity check it's not a boolean first. 

With the additional change to lower the initial rate to match the max rate if it's below the default, I refactored it to use if statements, which is hopefully more readable.